### ISSUE TITLE
iterator `return(..)` called twice with `for..of` + `break`

### ISSUE THREAD

**Issue by getify** [babel/babel#985]
Simple code to illustrate:

```js
var o = {
  [Symbol.iterator]() {
    return {
      next() { return { value: 1, done: false }; },
      return() { console.log("return'd"); }
    }
  }
}

for (var v of o) {
   break;
}
```

[This code in babel](http://babeljs.io/repl/#?experimental=false&playground=false&evaluate=true&loose=false&spec=true&code=var%20o%20%3D%20%7B%0A%20%20%5BSymbol.iterator%5D()%20%7B%0A%20%20%20%20return%20%7B%0A%20%20%20%20%20%20next()%20%7B%20return%20%7B%20value%3A%201%2C%20done%3A%20false%20%7D%3B%20%7D%2C%0A%20%20%20%20%20%20return()%20%7B%20console.log(%22return'd%22)%3B%20%7D%0A%20%20%20%20%7D%0A%20%20%7D%0A%7D%0A%0Afor%20(var%20v%20of%20o)%20%7B%0A%20%20%20break%3B%0A%7D) produces `"return'd"` twice.

By my understanding of the spec, `return()` should be called only once here, at the `break`, and not again at the end of the loop. IOW, it's called at the normal end of the loop only if it wasn't called earlier in abnormal termination. Am I correct, @allenwb?

**Comment by allenwb** [comment]
Yes, it should only be called once.
http://people.mozilla.org/~jorendorff/es6-draft.html#sec-runtime-semantics-forin-div-ofbodyevaluation-lhs-stmt-iterator-lhskind-labelset  Step 5.n.i

**Comment by sebmck** [comment]
Oddly enough this was a regression from making `for...of` **more** spec compliant. Fixed as of 4.7.5, thanks!

**Comment by getify** [comment]
:)
